internal/signer/linux/pkcs11/pkcs11.go

```diff
@@ -139,3 +144,24 @@ func (k *Key) Public() crypto.PublicKey {
 func (k *Key) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
 	return k.signer.Sign(nil, digest, opts)
 }
+
+func (k *Key) EncryptRSA(hash hash.Hash, data []byte) ([]byte, error) {
+	publicKey := k.Public()
+	rsaPubKey := publicKey.(*rsa.PublicKey)
+	return rsa.EncryptOAEP(hash, rand.Reader, rsaPubKey, data, nil)
+}
+
+func (k *Key) EncryptRSAGoPKCS11(data []byte) ([]byte, error) {
+	publicKeyFilter := pkcs11.Filter{pkcs11.ClassPublicKey,""}
+	pubObjs, err := (k.slot).Objects(publicKeyFilter)
+	pubObj := pubObjs[0]
+	if err != nil {
+		return nil, fmt.Errorf("encryptRSAGoPKCS11 public key error, %v", err)
+	}
+	k.privKey = pkcs11.WithPublicKeyHandle(k.privKey, pubObj)
+	return pkcs11.Encrypt(k.privKey, data)
+}
+
+func (k *Key) DecryptRSAGoPKCS11(encryptedData []byte) ([]byte, error) {
+	return pkcs11.Decrypt(k.privKey, encryptedData)
+}
```

Comment:
This code should be generic. You can see that the PKCS#11 module is generic, but this API leaks PKCS11 and RSA